lekce/3/javascript.md

```diff
@@ -1,73 +1,299 @@
+    </p>
+    <p>
+        Oproti tomu Python pracuje vždy s časem na serveru. Když pojedou naše stránky na serveru, který se nachází v ČR a má nastavený středoevropský čas, budeme všem uživatelům na světě ukazovat stále stejný, středoevropský čas.
+    </p>
+</details>
+
+---
+
+### Datové struktury
 
 ----
 
-Vygenerovat v Pythonu JSON, říct si o něj něj javascriptem a ty data nějak umístit.
+JavaScript nemá seznamy a slovníky. Místo toho má [pole][] (angl. _array_) a [objekty][] (angl. _object_).
+
+[pole]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections
+[objekty]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects
+
+```js
+var clothes = [
+    'shoes',
+    't-shirt',
+    'hat',
+    'skirt',
+    'socks',
+];
+console.log(clothes[1]); // vypíše t-shirt
+
+var angryBirdsScore = {
+    'veronika': 6810,
+    'dan': 4245,
+    'zuzka': 29233,
+    'honza': 3824,
+};
+console.log(angryBirdsScore['honza']); // vypíše 3824
+```
+
+Jak je vidět, na první pohled vypadají stejně. Občas se ale chovají trochu jinak, takže je dobré vždy ověřit, co přesně s nimi můžeš dělat a jak.
+
+Například pokud název vlastnosti objektu neobsahuje žádné speciální znaky, můžeš ho zapisovat i bez uvozovek a přistupovat k nim přes tečku:
+
+```js
+var population = {
+  karvina: 65141,
+  ostrava: 294200,
+  'frydek-mistek': 56945,
+};
+console.log(population.karvina); // vypíše 65141
+console.log(population['ostrava']); // 294200
+console.log(population['frydek-mistek']); // 56945
+```
 
 ---
 
-## Nějaký hotový řešení typu lightbox
+### Připojení JS
+
+----
+
+Podobně jako CSS, i JavaScript můžeš uložit do souboru a připojit k webové stránce v jejím HTML. Vytvoř si do prázdné složky jednoduchou webovou stránku, pojmenovanou třeba `index.html`:
+
+```html
+<!DOCTYPE HTML>
+<html>
+    <head>
+        <title>PyLadies</title>
+        <script src="hello.js"></script>
+    </head>
+    <body>
+        <h1>PyLadies</h1>
+        <p>PyLadies jsou <strong>nejlepší</strong>!</p>
+    </body>
+</html>
+```
 
+Nyní si vedle ní vytvoř soubor `hello.js`:
 
-Knihovna se používá takhle
+```js
+console.log('Hello!');
+```
 
-nalinkuješ js, css
+Když si stránku otevřeš v prohlížeči a podíváš se do konzole, uvidíš, že se v ní objevilo `Hello!`. Prohlížeč si spolu se stránkou stáhl i tvůj soubor s JavaScriptem a spustil jej jako program.
 
-nějak zavoláš 
+Úvod lekce zmiňoval animace a další zábavné věci. Pozdravy vypsané do konzole, přiznejme si to, ostřílenou PyLady už moc nenadchnou. Zkus tedy místo `console.log()` použít funkci `window.alert()`:
 
-nebo dokonce jen napíšeš data atributy a knihovna si to najde a zpracuje
+```js
+window.alert('Hello!');
+```
+
+Funkce `window.alert()` vytvoří vyskakovací tabulku s textem, který ji předáme. Uživatel bude muset tabulku potvrdit tlačítkem <button>OK</button>, jinak nezmizí.
```

Comment:
To sem zapomněl změnit, díky. Ještě jsem chtěl přidat dialogové, ale asi bych to nekomplikoval.